<?xml version="1.0" encoding="utf-8"?>
<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>
  
  <!-- Weights folder -->
  <arg name="yolo_model/folder" default="$(find robot_vision_openvino)/models/yolov3"/>

  <!-- Config and weights folder. -->
  <rosparam file="$(find robot_vision_openvino)/config/ros.yaml" command="load" ns="yolo_ros" />
  <rosparam file="$(find robot_vision_openvino)/config/yolov3.yaml" command="load" ns="yolo_ros" />

  <!-- Start openvino yolo node -->
  <node pkg="robot_vision_openvino" type="yolo_ros" name="yolo_ros" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="yolo_model/folder" value="$(arg yolo_model/folder)" />
  </node>

  <!-- Node to extract image names from YOLO output -->
  <node pkg="image_name_extractor" type="extract_yolo_image_names.py" name="extract_yolo_image_names" output="screen">
    <param name="input_yolo_topic" value="/yolo_ros/output_topic"/> <!-- Adjust based on actual YOLO output topic -->
    <param name="output_image_name_topic" value="/yolo_image_names"/> <!-- Publishes image names here -->
  </node>

  <!-- Optionally, you can republish images if needed -->
  <!--<node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
